lib/core/src/wallet.rs

```diff
@@ -354,35 +357,46 @@ impl OnchainWallet for LiquidOnchainWallet {
 
     /// Perform a full scan of the wallet
     async fn full_scan(&self) -> Result<(), PaymentError> {
-        let mut wallet = self.wallet.lock().await;
-        let electrum_url = ElectrumUrl::new(&self.config.liquid_electrum_url, true, true)?;
-        let mut electrum_client = ElectrumClient::new(&electrum_url)?;
+        let full_scan_started = Instant::now();
+
+        // create electrum client if doesn't already exist
+        let mut electrum_client = self.electrum_client.lock().await;
+        if electrum_client.is_none() {
+            let electrum_url = ElectrumUrl::new(&self.config.liquid_electrum_url, true, true)?;
+            *electrum_client = Some(ElectrumClient::new(&electrum_url)?);
+        }
+        let client = electrum_client.as_mut().unwrap();
```

Comment:
Maybe avoid the unwrap?
```suggestion
        let client = electrum_client.as_mut()?;
```

Reply:
Fixed